### ISSUE TITLE
question: recommended approach to working with large datasets and multiple queries

### ISSUE THREAD

**Issue by oocx** [datorama/akita#132]
Hi,

I'm new to akita and state management libraries in general. I'm currently investigating how to add state management to an existing Angular application. My application is a spare time project that is used to manage a list of starts for a gliding club.

Currently, my app gets data from a backend api (asp.net core + SQL Server) that has all kinds of different queries, for example "get all starts for a specific year", "get starts by pilot", "get starts by plane", "get all starts for the current user" and so on. A user can also add new starts and modify / delete starts.

I don't want to download unnecessary starts to the client to avoid large downloads of data that is never needed, so loading all starts for all years/pilots/planes etc. in advance is not an option, I need to download only the minimum required data.

So basically, I have several "query starts by X". If the app already has the results for "query by X", then I would return the version from the akita store, otherwise I would query the server.

As far as I understand akita so far, I will need to duplicate all the queries I already have in my API on the server as akita queries on the client. Is that correct, or is there a better approach that would not require duplication of all my queries? Having to maintain the same queries in two different places that even don't use the same programming language (it's an SQL ... FOR JSON query on the server) might make maintaining the app much more difficult in the long run.

What's the recommended approach for such a scenario?

Thanks,
Mathias

**Comment by NetanelBasal** [comment]
Hey, @oocx I would love to help. First, if you can add a simple demo of the API response, it would be great. Second, please join Akita's Gitter room so we can continue the discussion there. Thanks.
